app/backend/app.py

```diff
@@ -102,12 +114,23 @@ async def content_file(path: str):
         path_parts = path.rsplit("#page=", 1)
         path = path_parts[0]
     logging.info("Opening file %s", path)
-    blob_container_client = current_app.config[CONFIG_BLOB_CONTAINER_CLIENT]
+    blob_container_client: ContainerClient = current_app.config[CONFIG_BLOB_CONTAINER_CLIENT]
     try:
         blob = await blob_container_client.get_blob_client(path).download_blob()
     except ResourceNotFoundError:
-        logging.exception("Path not found: %s", path)
-        abort(404)
+        logging.info("Path not found in general Blob container: %s", path)
+        if current_app.config[CONFIG_USER_UPLOAD_ENABLED]:
+            try:
+                user_oid = auth_claims["oid"]
+                user_blob_container_client = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+                user_directory_client: FileSystemClient = user_blob_container_client.get_directory_client(user_oid)
```

Comment:
try:
                    await user_directory_client.get_directory_properties()
                except ResourceNotFoundError:
                    # The directory does not exist, so create it
                    logging.info(f"Creating directory for user {user_oid}")
                    await user_directory_client.create_directory()

Reply:
Huh, the reference for get_directory_client says that it need not exist:
https://learn.microsoft.com/en-us/python/api/azure-storage-file-datalake/azure.storage.filedatalake.filesystemclient?view=azure-python#azure-storage-filedatalake-filesystemclient-get-directory-client

But perhaps it's wrong or I misunderstood it. I'll delete my directory and test again, looks like I'll need that code. Thanks! @zedhaque I think you ran into this as well.

Reply:
%s not %S :)

Reply:
Thank you! I brought in that code, added a test, and filed an issue on the SDK re the confusing documentation: https://github.com/Azure/azure-sdk-for-python/issues/34939